nmdc/messages.go

```diff
@@ -639,7 +649,9 @@ func (m *MyInfo) unmarshalTag(tag []byte) error {
 				other[name] = value
 			}
 		} else {
-			return fmt.Errorf("unknown field in tag: %q", field)
+			if len(field) != 0 {
+				return fmt.Errorf("unknown field in tag: %q", field)
+			}
```

Comment:
What is the "else" condition here? I think you should add something like:
```golang
if len(field) == 0 {
  continue
}
```
to the beginning of the loop and remove `if len(field) != 0 {` from here, leaving only an error.